frontend/src/streamlist/StreamList.jsx

```diff
@@ -1,19 +1,41 @@
 import React from "react";
 import { Box, Container } from "@material-ui/core";
 import PropTypes from "prop-types";
-import * as useStyles from "../DefaultMakeStylesTheme";
 import StreamsTable from "./StreamsTable";
 
-export default function StreamList(props) {
-  const { dataSource } = props;
-  return (
-    <Container>
-      <Box className="headerAreaUnderline">
-        <div className="title">Current Streams</div>
-      </Box>
-      <StreamsTable classes={useStyles} dataSource={dataSource} />
-    </Container>
-  );
+export default class StreamList extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      streams: []
+    };
+    this.dataSource = props.dataSource;
+    this.handleStreamsChange = this.handleStreamsChange.bind(this);
+  }
+
+  componentDidMount() {
+    this.dataSource.getAllStreams(this.handleStreamsChange);
+  }
+
+  handleStreamsChange(streams) {
+    this.setState({
+      streams
+    });
+  }
+
+  render() {
+    const { streams } = this.state;
+    return (
+      <>
+        <Container>
+          <Box className="headerAreaUnderline">
+            <div className="title">Current Streams</div>
+          </Box>
+          <StreamsTable streamDetails={streams} />
```

Comment:
i'm not a huge fan of `streamDetails` as a prop name here when we're talking about an array of several streams. if you want to be consistent with the way we did the devices tables, just `streams` would be a good refactor

Reply:
Will change that.